python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py

```diff
@@ -527,28 +528,116 @@ def is_graph_backed_asset(self) -> bool:
-            )
+    async def get_materialization_status_for_asset_health(self, graphene_info: ResolveInfo):
+        if self._asset_node_snap.partitions is not None:  # isPartitioned
+            # TODO - maybe can do this from the asset record? that might not re-gen the cache though
+            partition_stats = self.resolve_partitionStats(graphene_info)
+            if partition_stats is None:
+                return GrapheneAssetHealthStatus.UNKNOWN
+            if partition_stats.numFailed > 0:
+                return GrapheneAssetHealthStatus.DEGRADED
+            if (
+                partition_stats.numPartitions
+                > partition_stats.numFailed
+                + partition_stats.numMaterialized
+                + partition_stats.numMaterializing
+            ):
+                # some partitions have never been materialized
+                return GrapheneAssetHealthStatus.WARNING
+            return GrapheneAssetHealthStatus.HEALTHY
+
+        asset_record = await AssetRecord.gen(graphene_info.context, self._asset_node_snap.asset_key)
+        if asset_record is None:
+            return GrapheneAssetHealthStatus.UNKNOWN
+        asset_entry = asset_record.asset_entry
+
+        if graphene_info.context.instance.can_read_failure_events_for_asset(
+            self._asset_node_snap.asset_key
         ):
-            # compute materialization status how we do today by checking the latest run the asset was materialized in (or using asset status cache for partitioned)
-            pass
+            # compute the status based on the asset key table
+
+            # if asset_entry.last_planned_materialization_storage_id is None:
+            #     return GrapheneAssetHealthStatus.UNKNOWN
+            #
+            # max_terminal_event_storage_id = max(asset_entry.last_materialization_storage_id, asset_entry.last_failed_to_materialize_storage_id)
+
+            # if last_planned_materialization_storage_id > max_terminal_event_storage_id:
+            #     # need to check if the run was canceled or deleted
+            #     run_record = await RunRecord.gen(
+            #         graphene_info.context,
+            #         check.not_none(asset_entry.last_planned_materialization_run_id),
+            #     )
+            #     if run_record is None:
+            #         # run was deleted TODO - should we fall back onto the previous materialization status?
+            #         return GrapheneAssetHealthStatus.UNKNOWN
+            #     if run_record.dagster_run.status == DagsterRunStatus.CANCELED:
+            #         # TODO - should we fall back onto the previous materialization status?
+            #         return GrapheneAssetHealthStatus.UNKNOWN
+            #     if run_record.dagster_run.status in IN_PROGRESS_RUN_STATUSES:
+            #         # TODO - should we fall back onto the previous materialization status?
+            #         return GrapheneAssetHealthStatus.UNKNOWN
+
+            if (
```

Comment:
depending on the outcome of https://github.com/dagster-io/internal/pull/14728 - may need to update this computation to ignore the latest failed materialization if it has a skip reason